homeassistant/components/energenie/config_flow.py

```diff
@@ -0,0 +1,53 @@
+"""Config flow for Energenie integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import HomeAssistantError
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Energenie Socket"): str,
+        vol.Required("socket_number", default=1): int,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for energenie."""
+
+    async def in_range(self, socket_number):
+        """Verify that input is between the range of 1 and 4."""
+        if 1 <= socket_number <= 4:
+            return True
+        raise InvalidRange
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await self.in_range(user_input["socket_number"])
+                await self.async_set_unique_id(f'energenie-socket-{user_input["socket_number"]}')
```

Comment:
We don't need to include the domain etc in the unique id for the config entry. I'm not sure that socket number qualifies as a unique id.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

Reply:
If we can't use the unique id api of the config entry, we can still deduplicate config entries, by using this helper:
https://github.com/home-assistant/core/blob/0327d0b6db4724a1229c209ba61d4da498d94d7a/homeassistant/config_entries.py#L1128-L1132